frontend/packages/gitops-plugin/src/components/history/GitOpsDeploymentHistory.tsx

```diff
@@ -0,0 +1,153 @@
+    const params = new URLSearchParams(location.search);
+    _.map(filterKeys, (f) => {
+      const vals = params.get(f);
+      if (vals) {
+        rowFiltersFromURL.push(...vals.split(','));
+      }
+    });
+    return { rowFiltersFromURL };
+  })();
+
+  const historyBaseURI = `/api/gitops/history/environment`;
+  const [historyData, setHistoryData] = React.useState<GitOpsHistoryData[]>(null);
+  const [error, setError] = React.useState<string>(null);
+  React.useEffect(() => {
+    let ignore = false;
+    const getHistory = async () => {
+      if (!_.isEmpty(envs) && applicationBaseURI) {
+        let arrayHistory;
+        try {
+          arrayHistory = await Promise.all(
+            _.map(envs, (env) =>
+              getEnvData(historyBaseURI, historyBaseURI, env, applicationBaseURI),
+            ),
+          );
+          arrayHistory = arrayHistory?.flat(1);
+        } catch (err) {
+          if (err instanceof Error) {
+            if (err.name === 'HttpError' && err.message === 'Not Found') {
+              setError(
+                t(
+                  'gitops-plugin~The history cannot be obtained due to an HTTP Not Found Error. This could mean that the GitOps Operator needs to be upgraded to the latest version or the GitOps cluster pod is not running.',
+                ),
+              );
+            } else {
+              setError(
+                t(
+                  'gitops-plugin~The history cannot be obtained due to an error. Check the GitOps cluster pod log for any errors.',
+                ),
+              );
+            }
+          }
+        }
+        if (ignore) return;
+        setHistoryData(arrayHistory);
+      }
+    };
+    getHistory();
+    return () => {
+      ignore = true;
+    };
+  }, [applicationBaseURI, envs, historyBaseURI, historyData, t]);
+
+  let filteredData: GitOpsHistoryData[];
+  if (historyData) {
+    filteredData = new Array<GitOpsHistoryData>();
+    if (selectedRowFilters.length > 0) {
+      historyData.forEach((history) => {
+        selectedRowFilters.forEach((filter) => {
+          if (history.environment === filter) {
+            filteredData.push(history);
+          }
+        });
+      });
+    } else {
+      filteredData = historyData.slice();
+    }
+  }
+
+  return (
+    <div className="odc-gitops-history-list">
+      <div>
+        <FilterToolbar
+          data={historyData}
+          reduxIDs={['gitops-environments']}
+          hideNameLabelFilters
+          rowFilters={envRowFilters}
+        />
+      </div>
+      {!historyData && !error ? (
+        <LoadingBox />
+      ) : error ? (
+        <GitOpsEmptyState emptyStateMsg={error} />
+      ) : emptyStateMsg ? (
+        <GitOpsEmptyState emptyStateMsg={emptyStateMsg || t('gitops-plugin~No history')} />
+      ) : (
+        <Table
+          data={filteredData}
+          aria-label={t('gitops-plugin~Deployment history')}
+          Header={GitOpsDeploymentHistoryTableHeader()}
```

Comment:
After this https://github.com/openshift/console/pull/11439#discussion_r867685398 change
```suggestion
          Header={GitOpsDeploymentHistoryTableHeader}
```

Reply:
Thanks @vikram-raj, good catch. Done.